networkmanager/networkmanager_test.go

```diff
@@ -71,542 +120,924 @@ func TestMain(m *testing.M) {
-		t.Errorf("Error: %s", err)
+	if in != 160 && out != 160 {
+		t.Fatal("Invalid traffic")
 	}
-}
 
-func TestNetworkDNS(t *testing.T) {
-	t.Cleanup(func() {
-		_ = killOCIContainer("service0")
-		_ = manager.DeleteNetwork("network0")
-		_ = manager.DeleteNetwork("network1")
-	})
+	for i := 0; i < 3; i++ {
+		in, out, err = manager.GetInstanceTraffic("instance1")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if in != testData[i].expectedInputTraffic && out != testData[i].expectedOutputTraffic {
+			t.Fatal("Invalid traffic")
+		}
+	}
 
-	container0Path := path.Join(tmpDir, "service0")
+	iptableInterface.disableResetMonitoringTraffic = false
 
-	if err := createOCIContainer(container0Path, "service0", []string{"sleep", "infinity"}); err != nil {
-		t.Fatalf("Can't create service container: %s", err)
+	in, out, err = manager.GetInstanceTraffic("instance1")
+	if err != nil {
+		t.Fatal(err)
 	}
 
-	if err := manager.AddServiceToNetwork("service0", "network0", networkmanager.NetworkParams{
-		Hostname: "myhost",
-		Aliases:  []string{"alias1"},
-	}); err != nil {
-		t.Fatalf("Can't add service to network: %s", err)
+	if in != 0 && out != 0 {
+		t.Fatal("Invalid traffic")
 	}
 
-	go runOCIContainer(container0Path, "service0") // nolint:errcheck
+	iptableInterface.disableResetMonitoringTraffic = true
 
-	container1Path := path.Join(tmpDir, "service1")
+	for i := 3; i < len(testData); i++ {
+		in, out, err = manager.GetInstanceTraffic("instance1")
+		if err != nil {
+			t.Fatal(err)
+		}
 
-	// Other SP network
+		if in != testData[i].expectedInputTraffic && out != testData[i].expectedOutputTraffic {
+			t.Fatal("Invalid traffic")
+		}
+	}
 
-	hostNames := []string{
-		"service0.network0",
-		"myhost.network0",
-		"alias1.network0",
+	if err := manager.SetTrafficPeriod(networkmanager.YearPeriod + 1); err == nil {
+		t.Fatal("Should be an error: unexpected period")
 	}
 
-	for _, hostName := range hostNames {
-		if err := createOCIContainer(container1Path, "service1", []string{"ping", hostName, "-c1"}); err != nil {
-			t.Fatalf("Can't create service container: %s", err)
-		}
+	if err := manager.RemoveInstanceFromNetwork("instance1", "network0"); err != nil {
+		t.Fatalf("Can't remove instance from network: %s", err)
+	}
 
-		if err := manager.AddServiceToNetwork("service1", "network1", networkmanager.NetworkParams{
-			ResolvConfFilePath: path.Join(container1Path, "etc", "resolv.conf"),
-		}); err != nil {
-			t.Fatalf("Can't add service to network: %s", err)
-		}
+	if _, _, err := manager.GetInstanceTraffic("instance1"); err == nil {
+		t.Fatal("Should be an error")
+	}
 
-		if err := runOCIContainer(container1Path, "service1"); err != nil {
-			t.Errorf("Error: %s", err)
-		}
+	manager.Close()
 
-		if err := manager.RemoveServiceFromNetwork("service1", "network1"); err != nil {
-			t.Fatalf("Can't remove service from network: %s", err)
-		}
+	if _, _, err = manager.GetSystemTraffic(); err == nil {
+		t.Fatal("Should be error")
```

Comment:
For all `should be error` message add reason why. Also, not all `should be error` should be fatal.

Reply:
Done in all places